README.md
# Adibags Prior Expansions
Warcraft addon filter for AdiBags, miscellaneous grouping options for tradeskills, gear, other items.

## To Do
Priority group for 6 relics to combine for assault chest key

CHANGES.txt
04.23.19    v 0.2    Completed first pass at BoP/BoE toggle
04.24.19    v 0.21   Added prior expansion crafted gem filter
04.25.19    v 1.0    Added group for items to be opened such as emissary bonuses or lockboxes
04.26.19    v 1.01   Fixed Category labels
05.04.19    v 1.02   Fixed BoE/BoP ilevel check
05.26.19    v 1.03   Added colors to Category labels
06.25.19    v 1.04   Added seperate catetory for mounts
09.28.19    v 1.05   Included elemental and cooking mats in prior expansions filter
05.30.20    v 1.06   Fixed memory bloat from tooltips
06.01.20    v 1.07   Cleaned up constants
06.09.20    v 1.08   More code cleanup
06.18.20    v 1.1.0  Major code rewrite: removed string parsed arrays, replaced with name/value pairs with nested arrays. Changed strings to constants for translation as much as possible. Added pet battle, heirloom and cosmetic gear groups. Added '#' prefix to group labels for prior expansion items. Added ability to toggle color for consumable and tradegood labels. Localization added for German and Spanish. 
06.18.23    v 1.1.01 Additonal translations, additional item level filtering

PriorExpansionData.lua
local addonName, addon = ...	 
--	 
	
--<GLOBALS      	
local _G = _G      	
local GetLocale = _G.GetLocale      	
local pairs = _G.pairs      	
local rawset = _G.rawset      	
local setmetatable = _G.setmetatable      	
local tostring = _G.tostring      	
--GLOBALS>      	
      	
local L = setmetatable({}, {      	
  __index = function(self, key)    	
    if key ~= nil then  	
      --[===[@debug@	
	
      addon:Debug('Missing locale', tostring(key))	
      --@end-debug@]===]	
      rawset(self, key, tostring(key))	
    end  	
    return tostring(key)  	
  end,    	
})      	
addon.L = L      	
	
local locale = GetLocale()	
----enUS	 
	 
   L["BoP"] = true	 
   L["BoE"] = true	 
   L["BoA"] = true	 
   L["Open Me!"] = true	 
   L["Cosmetic"] = true	 
   L["Pet Battle"] = true	 
   L["Miscellaneous"] = true	 
   L["Enable prior expansion groups"] = true	 
   L["Check this if you want to group by prior expansion. Prior expansion group bag labels are light blue."] = true	 
   L["Equippable Item Sub-Groupings"] = true	 
   L["Select optional groupings for weapons and armor."] = true	 
   L["Bind on Equip Gear"] = true	 
   L["Check to group Bind on Equip armor and weapons from prior expansions."] = true	 
   L["Check to group Soulbound armor and weapons from prior expansions."] = true	 
   L["Check to group Legendaries from prior expansions."] = true	 
   L["Check to group Artifacts from prior expansions."] = true	 
   L["Check to group cosmetic items like tabards and costumes."] = true	 
   L["Optional miscellaneous item groupings"] = true	 
   L["Select groupings options for non-gear items."] = true	 
   L["Check to group Trade Goods by category (Herbs, Leather, etc.) from prior expansions."] = true	 
   L["Food, Drink & Potions"] = true	 
   L["Check to group Food, Drink, Potions, Elixirs and Flasks from prior expansions."] = true	 
   L["Battle Pet Items"] = true	 
   L["Check to group battle pet items."] = true	 
   L["Group Unopened Loot"] = true	 
   L["Check to group lockboxes, bonus caches and other loot containers. Yeah, it\'s not expansion related, but it\'s handy!"] = true	 
   L["Separate Mount Drops"] = true	 
   L["Check to group Mounts reins so you don't space if one dropped and keep grinding needlessly."] = true	 
   L["Prior Expansion Groups"] = true	 
   L["Other Settings"] = true	 
   L["Other Prior Expansion filter settings"] = true	 
   L["Colored labels"] = true	 
   L["Prior Expansion Groups"] = true	 
   L["Check to use colored labels for prior expansion tradegoods and consumables."] = true	 
	 
----deDE	 
if locale=="deDE" then 	 
   L["BoP"] = "BoP"	 
   L["BoE"] = "BoE"	 
   L["BoA"] = "BoA"	 
   L["Open Me!"] = "Öffne mich!"	 
   L["Cosmetic"] = "Kosmetisch"	 
   L["Pet Battle"] = "Haustierkämpfe"	 
   L["Miscellaneous"] = "Sonstiges"	 
   L["Enable prior expansion groups"] = "Aktiviere Gruppen für frühere Erweiterungen"	 
   L["Check this if you want to group by prior expansion. Prior expansion group bag labels are light blue."] = "Aktiviere diese Option, wenn du Gegenstände aus früheren Erweiterungen gruppieren möchtest. Die Gruppen aus früheren Erweiterungen werden in der Tasche hellblau angezeigt."	 
   L["Equippable Item Sub-Groupings"] = "Untergruppen für Ausrüstungsteile"	 
   L["Select optional groupings for weapons and armor."] = "Optionale Gruppen für Waffen und Rüstungsteile."	 
   L["Bind on Equip Gear"] = "Beim Anlegen gebundene Gegenstände"	 
   L["Check to group Bind on Equip armor and weapons from prior expansions."] = "Gruppiere beim Anlegen gebundene Rüstungsteile und Waffen aus früheren Erweiterungen."	 
   L["Check to group Soulbound armor and weapons from prior expansions."] = "Gruppiere seelengebundene Rüstungsteile und Waffen aus früheren Erweiterungen."	 
   L["Check to group Legendaries from prior expansions."] = "Gruppiere legendäre Gegenstände aus früheren Erweiterungen."	 
   L["Check to group Artifacts from prior expansions."] = "Gruppiere Artefakte aus früheren Erweiterungen."	 
   L["Check to group cosmetic items like tabards and costumes."] = "Gruppiere kosmetische Gegenstände, wie z.B. Wappenröcke oder Kostüme."	 
   L["Optional miscellaneous item groupings"] = "Optionale Gruppierung für sonstige Gegenstände"	 
   L["Select groupings options for non-gear items."] = "Gruppen für nicht-Ausrüstungsgegenstände."	 
   L["Check to group Trade Goods by category (Herbs, Leather, etc.) from prior expansions."] = "Gruppiere Handwerksmaterialien (Kräuter, Leder, usw.) nach ihrer Kategorie aus früheren Erweiterungen."	 
   L["Food, Drink & Potions"] = "Speis & Trank & Tränke"	 
   L["Check to group Food, Drink, Potions, Elixirs and Flasks from prior expansions."] = "Gruppiere Speis & Trank, Tränke, Elixiere und Fläschchen aus früheren Erweiterungen."	 
   L["Battle Pet Items"] = "Haustierkampfgegenstände"	 
   L["Check to group battle pet items."] = "Gruppiere Gegenstände für Haustierkämpfe."	 
   L["Group Unopened Loot"] = "Gruppiere ungeöffnete Beute"	 
   L["Check to group lockboxes, bonus caches and other loot containers. Yeah, it\'s not expansion related, but it\'s handy!"] = "Gruppiere Schließkassetten und andere Beutebehälter. Ja, es hat nichts mit Erweiterungen zu tun, aber es ist praktisch!"	 
   L["Separate Mount Drops"] = "Trenne Reittiere"	 
   L["Check to group Mounts reins so you don't space if one dropped and keep grinding needlessly."] = "Hebt Reittiere in den Taschen hervor, damit sie leichter gesehen werden."	 
   L["Prior Expansion Groups"] = "TBD"	 
   L["Other Settings"] = "Zusätzliche Einstellungen"	 
   L["Other Prior Expansion filter settings"] = "Zusätzliche Einstellungen für Filter"	 
   L["Colored labels"] = "Farbige Gruppen"	 
   L["Prior Expansion Groups"] = "Gruppen für frühere Erweiterungen"	 
   L["Check to use colored labels for prior expansion tradegoods and consumables."] = "Handwerksmaterialien und Verbrauchsgüter aus vergangenen Erweiterungen werden farbig angezeigt"	 
end	 
----esMX	 
if locale=="esMX" then 	 
   L["BoP"] = "BoP "	 
   L["BoE"] = "BoE "	 
   L["BoA"] = "BoA "	 
   L["Open Me!"] = "¡Ábreme!"	 
   L["Cosmetic"] = "Cosmético"	 
   L["Pet Battle"] = "Batalla de mascotas"	 
   L["Miscellaneous"] = "Diverso"	 
   L["Enable prior expansion groups"] = "Habilitar grupos de expansión anteriores"	 
   L["Check this if you want to group by prior expansion. Prior expansion group bag labels are light blue."] = "Marque esto si desea agrupar por expansión previa. Las etiquetas de bolsas del grupo de expansión anterior son de color azul claro."	 
   L["Equippable Item Sub-Groupings"] = "Subgrupos de artículos equipables"	 
   L["Select optional groupings for weapons and armor."] = "Seleccione agrupaciones opcionales para armas y armaduras."	 
   L["Bind on Equip Gear"] = "Ate en Equipar equipo"	 
   L["Check to group Bind on Equip armor and weapons from prior expansions."] = "Marque para agrupar Vincular en Equipar armaduras y armas de expansiones anteriores."	 
   L["Check to group Soulbound armor and weapons from prior expansions."] = "Marque para agrupar armaduras y armas ligadas al alma de expansiones anteriores."	 
   L["Check to group Legendaries from prior expansions."] = "Marque para agrupar legendarios de expansiones anteriores."	 
   L["Check to group Artifacts from prior expansions."] = "Marque para agrupar artefactos de expansiones anteriores."	 
   L["Check to group cosmetic items like tabards and costumes."] = "Marque para agrupar artículos cosméticos como tabardos y disfraces."	 
   L["Optional miscellaneous item groupings"] = "Grupos opcionales de artículos varios"	 
   L["Select groupings options for non-gear items."] = "Seleccione las opciones de agrupación para los artículos sin engranaje."	 
   L["Check to group Trade Goods by category (Herbs, Leather, etc.) from prior expansions."] = "Marque para agrupar mercancías comerciales por categoría (hierbas, cuero, etc.) de expansiones anteriores."	 
   L["Food, Drink & Potions"] = "Comida, bebida y pociones"	 
   L["Check to group Food, Drink, Potions, Elixirs and Flasks from prior expansions."] = "Marque para agrupar alimentos, bebidas, pociones, elixires y frascos de expansiones anteriores."	 
   L["Battle Pet Items"] = "Artículos de batalla de mascotas"	 
   L["Check to group battle pet items."] = "Marque para agrupar elementos de mascotas de batalla."	 
   L["Group Unopened Loot"] = "Grupo de botín sin abrir"	 
   L["Check to group lockboxes, bonus caches and other loot containers. Yeah, it\'s not expansion related, but it\'s handy!"] = "Marque para agrupar cajas de seguridad, cachés de bonificación y otros contenedores de botín. Sí, no está relacionado con la expansión, ¡pero es útil!"	 
   L["Separate Mount Drops"] = "Gotas de montaje separadas"	 
   L["Check to group Mounts reins so you don't space if one dropped and keep grinding needlessly."] = "Marque para resaltar Monturas para que sean más fáciles de ver en bolsas."	 
   L["Prior Expansion Groups"] = "Grupos de Axpansión anteriores"	 
   L["Prior Expansion Groups"] = "Grupos de Expansión Anteriores"	 
end	 
	
for k,v in pairs(L) do if v == true then L[k] = k end end	
	
addon.arrPetBattle = {	[1] = { itemId=122457, zoneId=14, qty=-1, label="" },--PetBattle: Ultimate Battle-Training Stone
	[2] = { itemId=98715, zoneId=14, qty=-1, label="" },--PetBattle: Marked Flawless Battle-Stone
	[3] = { itemId=116429, zoneId=14, qty=-1, label="" },--PetBattle: Flawless Battle-Training Stone
	[4] = { itemId=127755, zoneId=14, qty=-1, label="" },--PetBattle: Fel-Touched Battle-Training Stone
	[5] = { itemId=92675, zoneId=14, qty=-1, label="" },--PetBattle: Flawless Beast Battle-Stone
	[6] = { itemId=92681, zoneId=14, qty=-1, label="" },--PetBattle: Flawless Undead Battle-Stone
	[7] = { itemId=92677, zoneId=14, qty=-1, label="" },--PetBattle: Flawless Flying Battle-Stone
	[8] = { itemId=92678, zoneId=14, qty=-1, label="" },--PetBattle: Flawless Magic Battle-Stone
	[9] = { itemId=92679, zoneId=14, qty=-1, label="" },--PetBattle: Flawless Aquatic Battle-Stone
	[10] = { itemId=92683, zoneId=14, qty=-1, label="" },--PetBattle: Flawless Dragonkin Battle-Stone
	[11] = { itemId=92680, zoneId=14, qty=-1, label="" },--PetBattle: Flawless Mechanical Battle-Stone
	[12] = { itemId=92676, zoneId=14, qty=-1, label="" },--PetBattle: Flawless Critter Battle-Stone
	[13] = { itemId=116374, zoneId=14, qty=-1, label="" },--PetBattle: Beast Battle-Training Stone
	[14] = { itemId=116424, zoneId=14, qty=-1, label="" },--PetBattle: Aquatic Battle-Training Stone
	[15] = { itemId=92682, zoneId=14, qty=-1, label="" },--PetBattle: Flawless Humanoid Battle-Stone
	[16] = { itemId=116416, zoneId=14, qty=-1, label="" },--PetBattle: Humanoid Battle-Training Stone
	[17] = { itemId=116423, zoneId=14, qty=-1, label="" },--PetBattle: Undead Battle-Training Stone
	[18] = { itemId=116418, zoneId=14, qty=-1, label="" },--PetBattle: Critter Battle-Training Stone
	[19] = { itemId=116419, zoneId=14, qty=-1, label="" },--PetBattle: Dragonkin Battle-Training Stone
	[20] = { itemId=116420, zoneId=14, qty=-1, label="" },--PetBattle: Elemental Battle-Training Stone
	[21] = { itemId=92665, zoneId=14, qty=-1, label="" },--PetBattle: Flawless Elemental Battle-Stone
	[22] = { itemId=137391, zoneId=14, qty=-1, label="" },--PetBattle: Immaculate Aquatic Battle-Stone
	[23] = { itemId=116417, zoneId=14, qty=-1, label="" },--PetBattle: Mechanical Battle-Training Stone
	[24] = { itemId=116421, zoneId=14, qty=-1, label="" },--PetBattle: Flying Battle-Training Stone
	[25] = { itemId=137393, zoneId=14, qty=-1, label="" },--PetBattle: Immaculate Critter Battle-Stone
	[26] = { itemId=116422, zoneId=14, qty=-1, label="" },--PetBattle: Magic Battle-Training Stone
	[27] = { itemId=137387, zoneId=14, qty=-1, label="" },--PetBattle: Immaculate Dragonkin Battle-Stone
	[28] = { itemId=137388, zoneId=14, qty=-1, label="" },--PetBattle: Immaculate Humanoid Battle-Stone
	[29] = { itemId=137390, zoneId=14, qty=-1, label="" },--PetBattle: Immaculate Mechanical Battle-Stone
	[30] = { itemId=137389, zoneId=14, qty=-1, label="" },--PetBattle: Immaculate Undead Battle-Stone
	[31] = { itemId=137392, zoneId=14, qty=-1, label="" },--PetBattle: Immaculate Magic Battle-Stone
	[32] = { itemId=137394, zoneId=14, qty=-1, label="" },--PetBattle: Immaculate Beast Battle-Stone
	[33] = { itemId=137395, zoneId=14, qty=-1, label="" },--PetBattle: Immaculate Elemental Battle-Stone
	[34] = { itemId=137396, zoneId=14, qty=-1, label="" },--PetBattle: Immaculate Flying Battle-Stone
	[35] = { itemId=86143, zoneId=14, qty=-1, label="" },--PetBattle: Pet Battle Bandage
	[36] = { itemId=163036, zoneId=14, qty=-1, label="" }, } --PetBattle: Polished Pet Charm
--	 
addon.maxExpansionIDs = {	 
  [1] = { classID=2, subClassID=-1, expLegion=163873, expBfA=0 },  -- Weapons, 	 
  [2] = { classID=4, subClassID=-1, expLegion=152118, expBfA=0 },  -- Armor, 	 
  [3] = { classID=0, subClassID=1, expLegion=152494, expBfA=0 },  -- Consumables, Potion	 
  [4] = { classID=0, subClassID=2, expLegion=151609, expBfA=0 },  -- Consumables, Elixir	 
  [5] = { classID=0, subClassID=3, expLegion=152638, expBfA=0 },  -- Consumables, Flask	 
  [6] = { classID=0, subClassID=5, expLegion=152592, expBfA=0 },  -- Consumables, Food	 
  [7] = { classID=7, subClassID=9, expLegion=152505, expBfA=0 },  -- Trade Goods, Herb	 
  [8] = { classID=7, subClassID=7, expLegion=152512, expBfA=0 },  -- Trade Goods, Metal & Stone	 
  [9] = { classID=7, subClassID=8, expLegion=152543, expBfA=0 },  -- Trade Goods, Cooking	 
  [10] = { classID=7, subClassID=5, expLegion=152576, expBfA=0 },  -- Trade Goods, Cloth	 
  [11] = { classID=7, subClassID=6, expLegion=152541, expBfA=0 },  -- Trade Goods, Leather	 
  [12] = { classID=7, subClassID=12, expLegion=152875, expBfA=0 },  -- Trade Goods, Enchanting	 
  [13] = { classID=7, subClassID=4, expLegion=153700, expBfA=0 },  -- Trade Goods, Jewelcrafting	 
  [14] = { classID=7, subClassID=-1, expLegion=152668, expBfA=0 },  -- Trade Goods, 	 
  [15] = { classID=8, subClassID=-1, expLegion=153700, expBfA=0 },  -- Item Enhancement, 	 
}	 
-- ZoneID Data	 
addon.arrZoneCodes = {	 
  [1]= { zGroup="Vale", zGroupIds={1530,1570,380,390} },	 
  [2]= { zGroup="Uldum", zGroupIds={1571,249,1527} },	 
  [3]= { zGroup="Vale/Uldum", zGroupIds={1571,249,1527,1530,1570,380} },	 
  [4]= { zGroup="Timeless Isle", zGroupIds={554} },	 
  [5]= { zGroup="Mechagon", zGroupIds={1462} },	 
  [6]= { zGroup="Nazjatar", zGroupIds={1355} },	 
  [7]= { zGroup="BfA Zones", zGroupIds={2161,2162,2160,2156,2103,2158,1504,1462} },	 
  [8]= { zGroup="Horrific Visions", zGroupIds={1469,1470, 1473} },	 
  [9]= { zGroup="Legion Missions", zGroupIds={647,648,626,717,734,735,24,702,695,747,719,831,883,887,726,739} },	 
  [10]= { zGroup="Heart Essence", zGroupIds={1473} },	 
  [11]= { zGroup="Garrison", zGroupIds={582,590,525} },	 
  [12]= { zGroup="REP: Nazjatar", zGroupIds={1355} },	 
  [13]= { zGroup="Reputation", zGroupIds={-1} },	 
  [14]= { zGroup="Pet Battle", zGroupIds={-1} } ,}	 
	
	
	

AdiBags_PriorExpansions.lua
--[[
AdiBags_PriorExpansions - Seperates items from current expansion from those from prior ones, an addition to Adirelle's fantastic bag addon AdiBags.
Copyright 2019 Ggreg Taylor
All rights reserved.
--]]
local addonName, addon = ...
local AdiBags = LibStub("AceAddon-3.0"):GetAddon("AdiBags")

local L = setmetatable({}, {__index = addon.L})
local setFilter = AdiBags:RegisterFilter("PriorExpansion", 93, 'ABEvent-1.0')
setFilter.uiName = L['Prior Expansion Groups']
setFilter.uiDesc = L['Group previous expansion items together.']

local currMinLevel = 201
local kCategory = L['Prior Expansion']
local kPfx = '|cff00ffff' 
local kPfxColor2 = '|cff4bf442' 
local kSfx = '|r'
local kPfxTradegoods = ''
local Ggbug = false
local debugBagSlot = {1,8}
local lookForId = 86143
local bagItemID
local PRIORITY_ITEM = L['Attention!']
AdiBags:SetCategoryOrder(PRIORITY_ITEM, 81)

local function Ggprint(...) 
  if lookForId == bagItemID and Ggbug == true then print(...) end
end

local function create()
  local tip, leftTip, rightTip = CreateFrame("GameTooltip"), {}, {}
  for x = 1,6 do
    local L,R = tip:CreateFontString(), tip:CreateFontString()
    L:SetFontObject(GameFontNormal)
    R:SetFontObject(GameFontNormal)
    tip:AddFontStrings(L,R)
    leftTip[x] = L
    rightTip[x] = R
  end
  tip.leftTip = leftTip
  tip.rightTip = rightTip
  return tip
end
local tooltip = tooltip or create()


function setFilter:OnInitialize()
  self.db = AdiBags.db:RegisterNamespace('PriorExpansion', {
    profile = { enable = true ,
    enableMats = true,
    enableBoE = true,
    enableBoP = true,
    enableLegendaries = true,
    enableArtifacts = true,
    enableMats = true,
    enableConsumables = true,
    enableToOpen = true,
    enableMounts = true,
    enableCosmetic = false,
    enablePetGear = false,
    enableColoredLabels = true,
  },
    char = {  },
  })
end
function setFilter:Update()
  self:SendMessage('AdiBags_FiltersChanged')
end
function setFilter:OnEnable()
  AdiBags:UpdateFilters()
end
function setFilter:OnDisable()
  AdiBags:UpdateFilters()
end
function setFilter:GetOptions()
  return {
    enable = {
      name = L['Enable prior expansion groups'],
      desc = L['Check this if you want to group by prior expansion. Prior expansion group bag labels are light blue.'],
      width = 'double',
      type = 'toggle',
      order = 25,
    },
    priorExpansionGear = {
      name = L['Equippable Item Sub-Groupings'],
      type = L['group'],
      inline = true,
      order = 65,
      args = {
        _desc = {
          name = L['Select optional groupings for weapons and armor.'],
          type = 'description',
          order = 10,
        },
        enableBoE = {
          name = L['Bind on Equip Gear'],
          desc = L['Check to group Bind on Equip armor and weapons from prior expansions.'],
          type = 'toggle',
          order = 20,
        },
        enableBoP = {
          name = ITEM_SOULBOUND,
          desc = L['Check to group Soulbound armor and weapons from prior expansions.'],
          type = 'toggle',
          order = 30,
        },
        enableLegendaries = {
          name = ITEM_QUALITY5_DESC,
          desc = L['Check to group Legendaries from prior expansions.'],
          type = 'toggle',
          order = 33,
        },
        enableArtifacts = {
          name = ITEM_QUALITY6_DESC,
          desc = L['Check to group Artifacts from prior expansions.'],
          type = 'toggle',
          order = 40,
        },
        enableHeirlooms = {
          name = ITEM_QUALITY7_DESC,
          desc = L['Check to group Heirloom items.'],
          type = 'toggle',
          order = 40,
        },
        enableCosmetic = {
          name = L['Cosmetic'],
          desc = L['Check to group cosmetic items like tabards and costumes.'],
          type = 'toggle',
          order = 40,
        },
      }
    },
    priorExpansionItems = {
      name = L['Optional miscellaneous item groupings'],
      type = L['group'],
      inline = true,
      order = 65,
      args = {
        _desc = {
          name = L['Select groupings options for non-gear items.'],
          type = 'description',
          order = 10,
        },
        enableMats = {
          name = BAG_FILTER_TRADE_GOODS,
          desc = L['Check to group Trade Goods by category (Herbs, Leather, etc.) from prior expansions.'],
          type = 'toggle',
          order = 33,
        },
        enableConsumables = {
          name = L['Food, Drink & Potions'],
          desc = L['Check to group Food, Drink, Potions, Elixirs and Flasks from prior expansions.'],
          type = 'toggle',
          order = 40,
          
        },
        enablePetGear = {
          name = L['Battle Pet Items'],
          desc = L['Check to group battle pet items.'],
          type = 'toggle',
          order = 45,
          
        },
        
        enableToOpen = {
          name = L['Group Unopened Loot'],
          desc = L['Check to group lockboxes, bonus caches and other loot containers. Yeah, it\'s not expansion related, but it\'s handy!'],
          type = 'toggle',
          order = 50,
        },
        enableMounts = {
          name = L['Separate Mount Drops'],
          desc = L['Check to group Mounts reins so you don\'t space if one dropped and keep grinding needlessly.'],
          type = 'toggle',
          order = 60,
        },
      }
    },
    priorExpansionOtherSettings = {
      name = L['Other Settings'],
      type = L['group'],
      inline = true,
      order = 70,
      args = {
        _desc = {
          name = L['Other Prior Expansion filter settings.'],
          type = 'description',
          order = 10,
        },
        enableColoredLabels = {
          name = kPfx .. L['Colored Labels'] .. kSfx,
          desc = L['Check to use colored labels for prior expansion tradegoods and consumables.'],
          type = 'toggle',
          order = 33,
        },
      }

    },
  }, AdiBags:GetOptionHandler(self, false, function() return self:Update() end)
end

local function isFromPriorExpansion(itemClassID, itemSubClassID, itemId)
  -- compare to addon.maxExpansionIDs array, if itemId less than expLegion # then return true, is prior
  for k, v in pairs(addon.maxExpansionIDs) do
    if v.classID == itemClassID and v.subClassID == itemSubClassID then
      if itemId < v.expLegion then
        return true
      else
        return false
      end
    end
  end
    -- else catch all for parts, elemental, other categories, subClassID -1 indicates check for all other subclasses
  for k, v in pairs(addon.maxExpansionIDs) do
    if v.classID == itemClassID and v.subClassID == -1 then
      if itemId < v.expLegion then
        return true
      else
        return false
      end
    end
  end
  return false
end

------------------------------------------------------------------------------
function setFilter:checkItem(itemId, dataArray)
  -- returns zoneId if itemId finds a match in the array otherwise null
  --itemId, zoneId, qty-1, label
  for id, info in pairs(dataArray) do
    if tonumber(itemId) == tonumber(info.itemId) then
      --if qty is a number and matched by item quantity then mark as labeled  
      if GetItemCount(itemId, true) == tonumber(info.qty) then return true, kPfx .. info.label .. kSfx, PRIORITY_ITEM end
   
      local isCurrent, zoneGroup = setFilter:isCurrentZone(info.zoneId) 
      zoneGroup = L[zoneGroup]
      return true, kPfx .. zoneGroup .. kSfx, zoneGroup
    end
  end
  return false, false, false
end

----------------------------------------
function setFilter:isCurrentZone(zoneId)
  for id, info in pairs(addon.arrZoneCodes) do 
    if tonumber(id) == tonumber(zoneId) then 
      for x = 1, #info.zGroupIds do
        if tonumber(info.zGroupIds[x]) == currZoneId then return true, info.zGroup end
      end -- end for x
      return false, info.zGroup 
    end
  end 
end

function setFilter:Filter(slotData)
  if (self.db.profile.enable == false) or (slotData.itemId == false) then return end
  if self.db.profile.enableColoredLabels == true then kPfxTradegoods = kPfx else kPfxTradegoods = '' end
  
  local itemName, itemLink, itemRarity, itemLevel, itemMinLevel, itemType, itemSubType, _, _, _, _, itemClassID, itemSubClassID, bindType, expacID, _, isCraftingReagent = GetItemInfo(slotData.link)
  local isWeaponOrArmor = false
  local isAzeriteGear = C_AzeriteEmpoweredItem.IsAzeriteEmpoweredItemByID(slotData.link)
  local isEquipmentSetItem, equipmentSetName = GetContainerItemEquipmentSetInfo(slotData.bag, slotData.slot)
  --[[
  and isEquipmentSetItem == false 
---]]  
  if itemClassID == LE_ITEM_CLASS_WEAPON or itemClassID == LE_ITEM_CLASS_ARMOR then isWeaponOrArmor = true end
  bagItemID = slotData.itemId
  if not itemMinLevel then itemMinLevel = 0 end

  if self.db.profile.enableMounts and itemType == MISCELLANEOUS and itemSubType == MOUNT  then return  kPfxColor2 ..'EEK!'.. kSfx, PRIORITY_ITEM end
  -- if it's in an item set don't filter it

  if self.db.profile.enableHeirlooms and isEquipmentSetItem == false and itemRarity == LE_ITEM_QUALITY_HEIRLOOM  then  return  ITEM_QUALITY7_DESC, ITEM_QUALITY7_DESC end --Heirlooms
  if self.db.profile.enableLegendaries and isEquipmentSetItem == false and itemRarity == LE_ITEM_QUALITY_LEGENDARY  and isWeaponOrArmor then  return  ITEM_QUALITY5_DESC, ITEM_QUALITY5_DESC end --legendaries
  if self.db.profile.enableArtifacts and isEquipmentSetItem == false and itemRarity == LE_ITEM_QUALITY_ARTIFACT  and isWeaponOrArmor  then  return  ITEM_QUALITY6_DESC, ITEM_QUALITY6_DESC end --Artifacts
  if self.db.profile.enableMats and itemClassID == LE_ITEM_CLASS_TRADEGOODS and itemSubClassID ~= 16 then -- Don't group old inscription stuff
    if isFromPriorExpansion(itemClassID, itemSubClassID, bagItemID) then return kPfxTradegoods .. '#' .. itemSubType .. kSfx, '#' .. itemSubType end
  end
  if self.db.profile.enablePetGear then -- group Pet Battle items based on IDs from arrPetBattle
    local itemFound, groupLabel, retCategory = setFilter:checkItem(bagItemID, addon.arrPetBattle) 
    if itemFound == true  then return L[groupLabel], retCategory end
  end

  --- Groups that require scanning tooltip
  tooltip:SetOwner(UIParent,"ANCHOR_NONE")
  tooltip:ClearLines()
  if slotData.bag == BANK_CONTAINER then
    tooltip:SetInventoryItem("player", BankButtonIDToInvSlotID(slotData.slot, nil))
  else
    tooltip:SetBagItem(slotData.bag, slotData.slot)
  end
  tipData = {}
  for x = 1,6 do tipData[x] = {tooltip.leftTip[x]:GetText(), tooltip.rightTip[x]:GetText()} end
  tooltip:Hide()
  tooltip:SetParent(nil)
  -- Pet battle items

  if self.db.profile.enableCosmetic then
    if C_Item.GetItemInventoryTypeByID(bagItemID) == 19 or (itemRarity == 1 and itemLevel ==1 and isWeaponOrArmor == true) then return L['Cosmetic'], L['Cosmetic'] end -- 19 is tabardItemType enum
    for x = 1, 6 do
      if tipData[x][2] == L['Cosmetic'] then return L['Cosmetic'], L['Cosmetic'] end
    end
  end  
  -- Filter consumables, put down here because it conflicts with pet items
  if self.db.profile.enableConsumables and (itemClassID==LE_ITEM_CLASS_CONSUMABLE or itemClassID ==LE_ITEM_CLASS_ITEM_ENHANCEMENT) then
    if  isFromPriorExpansion(itemClassID, itemSubClassID, bagItemID) then  return kPfxTradegoods .. '#' .. itemType ..kSfx, '#' .. itemType  end 
  end

  for x = 1,6 do
    -- Filter Old BoE Gear
    if self.db.profile.enableBoE and tipData[x][1] == ITEM_BIND_ON_EQUIP and itemLevel < currMinLevel  and isWeaponOrArmor == true then
      return  kPfx .. '#' .. ITEM_BIND_ON_EQUIP .. kSfx, 'Old BoE Gear'
    end
    -- Filter Old BoP Gear
    if self.db.profile.enableBoP and tipData[x][1] == ITEM_SOULBOUND and  isWeaponOrArmor == true and itemLevel < currMinLevel then
      return  kPfx .. '#'.. ITEM_SOULBOUND.. kSfx, 'Old BoP Gear'
    end
    -- Filter Lockboxes
    if self.db.profile.enableToOpen and (tipData[x][1] == ITEM_OPENABLE or tipData[x][1] == LOCKED) then
      return  kPfx ..L['Open Me!'].. kSfx, NEW
    end
  end

end
